tools/get_coverage_ros2_pkg.py

```diff
@@ -0,0 +1,116 @@
+        if ros2_repos.status_code != requests.codes.ok:
+            print('Failed to download ros2.repos file', file=sys.stderr)
+            sys.exit(-1)
+        with open(ros2_repos_path, 'wb') as file:
+            file.write(ros2_repos.content)
+
+        cmd = ['vcs', 'import', ros2_ws_path, '--shallow', '--retry', '5', '--input', ros2_repos_path]
+        try:
+            print('Getting ros2.repos sources to get packages source paths. Please wait')
+            subprocess.check_output(cmd)
+        except subprocess.CalledProcessError as e:
+            print(e.output, file=sys.stderr)
+            sys.exit(-1)
+
+        # Get relative path in ROS workspace to real the package source code
+        ros_package_path_in_ws = get_src_path(input_pkg, ros2_ws_path)
+        input_pkg_rel_path = os.path.relpath(ros_package_path_in_ws, ros2_ws_path).split(os.path.sep)
+
+        coverage_entries = r.json()['results']['children']
+        total_lines_under_testing = 0
+        total_lines_tested = 0
+
+        for e in coverage_entries:
+            if e['name'] == '.':
+                continue
+            # e has children, elements or name
+            entry_name = e['name'].replace("'", "")
+            # search for the Lines coverage method
+            lines_coverage = None
+            for coverage_method in e['elements']:
+                if coverage_method['name'] == 'Lines':
+                    lines_coverage = coverage_method
+            assert lines_coverage, 'Did not find Lines metric in coverage'
+
+            name_parts = entry_name.split('.')
+
+            if len(name_parts) == 1:
+                package_under_cov = name_parts[0]
+            elif name_parts[0].startswith('test'):
+                # no interest in test code
+                continue
+            elif name_parts[0].startswith('install'):
+                # integration/system testing, out by now
+                continue
+            elif name_parts[0].startswith('build'):
+                # in build the first part is always the ROS package name
+                package_under_cov = name_parts[1]
+            elif is_slice_in_list(input_pkg_rel_path, name_parts):
+                # source code: check if coverage entry contains exactly the source path
+                package_under_cov = input_pkg
+            else:
+                package_under_cov = 'NOT-FOUND'
+
+            if package_under_cov == input_pkg:
+                total_lines_under_testing += lines_coverage['denominator']
+                total_lines_tested += lines_coverage['numerator']
+                print(f" * {entry_name}"
+                      f" [{lines_coverage['ratio']:.2f}] --"
+                      f" {lines_coverage['numerator']:.0f}/{lines_coverage['denominator']:.0f}")
+
+        if total_lines_under_testing == 0:
+            print('Package not found: ' + input_pkg, file=sys.stderr)
```

Comment:
The message seems to not describe what was being tested.

Reply:
I changed it to say it does not found coverage lines for package 4d4d9fe